pkgs/development/libraries/freeimage/default.nix

```
@@ -1,48 +1,39 @@
-{ lib, stdenv, fetchurl, unzip, darwin }:
-
-# TODO: consider unvendoring various dependencies (libpng, libjpeg,
-# libwebp, zlib, ...)
+{ lib, stdenv, fetchzip, unzip, darwin, libtiff
+, libpng, zlib, libwebp, libraw, openexr, openjpeg
+, libjpeg, jxrlib, pkgconfig, fetchpatch }:
 
 stdenv.mkDerivation {
-  name = "freeimage-3.18.0";
+  pname = "freeimage";
+  version = "unstable-2020-07-04";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/freeimage/FreeImage3180.zip";
-    sha256 = "1z9qwi9mlq69d5jipr3v2jika2g0kszqdzilggm99nls5xl7j4zl";
+  src = fetchzip {
```

It appears that sourceforge purges generated snapshots from cache after some time. Trying to download this today resulted in:
```
  trying https://sourceforge.net/code-snapshots/svn/f/fr/freeimage/svn/freeimage-svn-r1859.zip
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (22) The requested URL returned error: 404 
```
I had to go to the svn gui at https://sourceforge.net/p/freeimage/svn/HEAD/tree/ and click "Download Snapshot" to make that link valid again.

Overall I suppose we can't trust sourceforge's snapshots to be reproducible. Since you used `fetchzip` instead of `fetchurl` this would not be cached on tarballs.nixos.org. I would suggest switching to `fetchsvn`:
```suggestion
  src = fetchsvn {
    url = "svn://svn.code.sf.net/p/freeimage/svn/";
    rev = "1859";
    sha256 = "1d94935aqbkb994nqkw7m8xcynyz9rm6k7k59igrbjak8b63qpi6";
  }
```

---

Funnily enough this problem [already happened once](https://github.com/NixOS/nixpkgs/pull/92708#discussion_r550430348). I didn't know you could access the svn repository like that, thanks.

---

> Funnily enough this problem [already happened once](https://github.com/NixOS/nixpkgs/pull/92708#discussion_r550430348). I didn't know you could access the svn repository like that, thanks.

I hope using the svn:// would address that